packages/app/lib/src/features/home/presentation/home/post.dart

```diff
@@ -133,6 +135,43 @@ class _PosterInfo extends StatelessWidget {
   // coverage:ignore-end
 }
 
+class _PostAvatar extends ConsumerWidget {
+  const _PostAvatar({
+    required this.post,
+    super.key,
+  });
+
+  final PostEntity post;
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final avatarService = ref.watch(avatarsProvider);
+    return FutureBuilder<MemoryImage>(
+      future: getAvatar(avatarService),
+      builder: (context, snapshot) {
+        if (snapshot.connectionState == ConnectionState.waiting) {
+          return const CircularProgressIndicator();
+        } else if (snapshot.hasError) {
+          return const Text('Error loading avatar');
+        } else {
+          return CircleAvatar(
+            backgroundImage: snapshot.data,
+          );
+        }
+      },
+    );
+  }
+
+  Future<MemoryImage> getAvatar(Avatars avatarService) async {
+    return MemoryImage(await avatarService.getInitials(name: post.authorName));
+  }
```

Comment:
This fetches them every frame. (well, flutter and images are weird, it might just be per every time it gets a new render, but still, should be globally memoized)

I forgot to share this with you: https://github.com/PSDTools/app/blob/2b793de1d796cb215a91a2b1767d95fbb80481cc/lib/features/auth/data/avatar_repository.dart
https://github.com/PSDTools/app/blob/2b793de1d796cb215a91a2b1767d95fbb80481cc/lib/features/auth/data/auth_repository.dart

I've been wanting to add a drawer, I might just copy all this code over.